src/games/strategy/triplea/ui/MapPanel.java

```diff
@@ -213,25 +213,11 @@ public void setRoute(final Route route) {
   public void setRoute(final Route route, final Point start, final Point end, final Image cursorImage) {
     if (route == null) {
       routeDescription = null;
-      SwingUtilities.invokeLater(new Runnable() {
-        @Override
-        public void run() {
-          repaint();
-        }
-      });
-      return;
-    }
-    final RouteDescription newVal = new RouteDescription(route, start, end, cursorImage);
-    if (routeDescription != null && routeDescription.equals(newVal)) {
```

Comment:
Are we sure we should remove this check? It appears that if the route is the same as the existing one then we don't repaint it which kind of makes sense.